Classes/Service/OpenIdConnectService.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Causal\Oidc\Service;
+
+use Causal\Oidc\AuthenticationContext;
+use InvalidArgumentException;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use TYPO3\CMS\Core\Configuration\ExtensionConfiguration;
+use TYPO3\CMS\Core\Http\ServerRequest;
+use TYPO3\CMS\Core\Http\Uri;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class OpenIdConnectService implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    protected OAuthService $OAuthService;
+
+    protected ?AuthenticationContext $authContext = null;
+
+    /**
+     * Global extension configuration
+     */
+    protected array $config;
+
+    public function __construct(OAuthService $OAuthService, array $config = [])
+    {
+        $this->OAuthService = $OAuthService;
+        $this->config = $config ?: GeneralUtility::makeInstance(ExtensionConfiguration::class)->get('oidc') ?? [];
+    }
+
+    public function generateAuthenticationContext(ServerRequest $request, array $authorizationUrlOptions = []): AuthenticationContext
+    {
+        if (empty($this->config['oidcClientKey'])
+            || empty($this->config['oidcClientSecret'])
+            || empty($this->config['oidcEndpointAuthorize'])
+            || empty($this->config['oidcEndpointToken'])
+        ) {
+            throw new InvalidArgumentException('Missing extension configuration', 1715775147);
+        }
+
+        $requestId = $this->getUniqueId();
+
+        $this->logger->debug('Generating OpenID Connect URI', ['request' => $requestId]);
+
+        if (!$this->authContext || $this->authContext->requestId !== $requestId) {
```

Comment:
The requestID check does not make any sense. It is a random number, it will not exist before.